pkg/volume/volume_template_engine.go

```diff
@@ -81,7 +81,7 @@ func NewCASVolumeEngine(
 	casConfigSC string,
 	casTemplate *v1alpha1.CASTemplate,
 	runtimeKey string,
-	runtimeVolumeValues map[string]interface{}) (volumeEngine *casVolumeEngine, err error) {
+	runtimeVolumeValues map[string]interface{}) (volumeEngine engine.CASCreator, err error) {
```

Comment:
I like this !
"Be conservative in what you send, be liberal in what you accept"

Reply:
The big issue here is we were returning a private struct on a public method. We could make the method itself private and then we don't need to change this.

The other option is to return a better interface to describe what the casVolumeEngine does.